flink-connectors/flink-connector-elasticsearch6/src/main/java/org/apache/flink/streaming/connectors/elasticsearch6/Elasticsearch6UpsertTableSink.java

```diff
@@ -189,6 +207,67 @@ protected ElasticsearchUpsertTableSinkBase copy(
 	// Helper classes
 	// --------------------------------------------------------------------------------------------
 
+	/**
+	 * This class implements {@link RestClientFactory}, used for es with authentication.
+	 */
+	static class AuthRestClientFactory implements RestClientFactory {
+
+		private String userName;
+
+		private String password;
+
+		private Integer maxRetryTimeout;
+
+		private String pathPrefix;
+
+		private transient CredentialsProvider credentialsProvider;
+
+		public AuthRestClientFactory(@Nullable String userName, @Nullable String password,
```

Comment:
`username` and `password` are not nullable, because you will use `AuthRestClientFactory` iff username and password are set.